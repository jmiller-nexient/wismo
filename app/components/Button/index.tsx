import React from 'react';
import classnames from 'classnames';

export type ButtonHtmlType = 'button' | 'reset' | 'submit';
export type ButtonTypes = 'primary' | 'secondary';

export interface IButtonProps {
  children?: React.ReactNode;
  className?: string;
  disabled?: boolean;
  htmlType?: ButtonHtmlType;
  id?: string;
  onClick?: () => void;
  type?: ButtonTypes;
}

const getClasses = (className?: string, type?: ButtonTypes): string =>
  classnames(className, {
    'dte-wismo-button': true,
    'dte-wismo-button-primary': !type || type === 'primary',
    'dte-wismo-button-secondary': type === 'secondary',
  });

const Button: React.FC<IButtonProps> = (props: IButtonProps): React.ReactElement => {
  const { children, className, disabled, htmlType, id, onClick, type } = props;

  const classes: string = getClasses(className, type);

  return (
    // eslint-disable-next-line react/button-has-type
    <button className={classes} disabled={disabled} id={id} onClick={onClick} type={htmlType}>
      {React.Children.toArray(children)}
    </button>
  );
};

export default Button;
