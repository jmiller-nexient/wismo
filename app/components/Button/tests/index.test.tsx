import React, { ReactElement } from 'react';
import renderer from 'react-test-renderer';

import Button from '..';

describe('<Button />', () => {
  it('should render correctly when it is primary', () => {
    const tree: ReactElement = renderer.create(
      <Button type="primary" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when it is primary and disabled', () => {
    const tree: ReactElement = renderer.create(
      <Button disabled type="primary" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when it is secondary', () => {
    const tree: ReactElement = renderer.create(
      <Button type="secondary" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when it is secondary and disabled', () => {
    const tree: ReactElement = renderer.create(
      <Button disabled type="secondary" />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
