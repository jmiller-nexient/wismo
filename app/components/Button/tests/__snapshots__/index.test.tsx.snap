// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render correctly when it is primary 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-primary"
/>
`;

exports[`<Button /> should render correctly when it is secondary 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-secondary"
/>
`;

exports[`<Button /> should render correctly when it is primary and disabled 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-primary"
  disabled={true}
/>
`;

exports[`<Button /> should render correctly when it is secondary and disabled 1`] = `
<button
  className="dte-wismo-button dte-wismo-button-secondary"
  disabled={true}
/>
`;
